datafusion/core/src/physical_plan/sorts/sort_preserving_merge.rs

```diff
@@ -354,17 +333,13 @@ impl SortPreservingMergeStream {
             .map(|_| VecDeque::new())
             .collect();
         tracking_metrics.init_mem_used(streams.iter().map(|s| s.mem_used).sum());
-        let wrappers = streams
-            .into_iter()
-            .map(|s| StreamWrapper::Stream(Some(s)))
-            .collect();
+        let wrappers = streams.into_iter().map(|s| s.stream.fuse()).collect();
```

Comment:
Is `fuse` the magic that avoids the need for the stream wrapper?

Reply:
It's the combination of this and using RecordBatchReceiverStream to convert the mpsc to a SendableRecordBatchStream

Reply:
`RecordBatchReceiverStream` is the thing I was missing